src/kdc/kdc_util.c

```diff
@@ -1623,14 +1636,22 @@ kdc_process_s4u2self_req(kdc_realm_t *kdc_active_realm,
         }
 
         *princ_ptr = princ;
-    } else if (is_local_tgt) {
-        /*
-         * The server is asking to impersonate a principal from another realm,
-         * using a local TGT.  It should instead ask that principal's realm and
-         * follow referrals back to us.
-         */
-        *status = "S4U2SELF_CLIENT_NOT_OURS";
-        return KRB5KDC_ERR_POLICY; /* match Windows error */
+    } else {
```

Comment:
`return 0` at the end of the `if`, and then you can de-indent everything in the `else` block.

Reply:
Yeah, that would be cleaner. Will do.